From 9f1d01baec6e3aaf3e834572834934aa76080e8b Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Wed, 22 Feb 2012 16:08:24 +0800
Subject: [PATCH 1084/2508] ENGR00174914 MX6x HDMI implement HDMI driver
 suspend/resume function

Added FB suspend/resume event process in HDMI driver.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/video/mxc_hdmi.c |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/drivers/video/mxc_hdmi.c b/drivers/video/mxc_hdmi.c
index c13bc70..ef3ebd3 100644
--- a/drivers/video/mxc_hdmi.c
+++ b/drivers/video/mxc_hdmi.c
@@ -2090,6 +2090,20 @@ static int mxc_hdmi_fb_event(struct notifier_block *nb,
 			mxc_hdmi_phy_disable(hdmi);
 		}
 		break;
+
+	case FB_EVENT_SUSPEND:
+		dev_dbg(&hdmi->pdev->dev,
+			"event=FB_EVENT_SUSPEND\n");
+		mxc_hdmi_phy_disable(hdmi);
+		break;
+
+	case FB_EVENT_RESUME:
+		dev_dbg(&hdmi->pdev->dev,
+			"event=FB_EVENT_RESUME\n");
+		if (hdmi->fb_reg && hdmi->cable_plugin)
+			mxc_hdmi_phy_init(hdmi);
+		break;
+
 	}
 	return 0;
 }
-- 
1.7.7.4

